### PR TITLE
Add default enchantment level for enchant sign

### LINKED ISSUE #1497: 1.9 or up Enchant is not available thru sign
EssentialsX version (`/essentials`): 2.0.1-b523

Server software (`/version`): git-Spitogt-65e8124-963b19c (MC: 1.12.1) (API 1.12.1-R0.1-SNAPSHOT)

EssentialsX config (if applicable): default config

Steps to replicate issue:
1) Put an enchant sign, on line 3 write `Mending`
2) Click Done.
3) It breaks.

### PR DESCRIPTION
**Pull request by JRoy**
Defaults the enchantment level on signs to `1` is one is not provided on the sign.

Closes #1497 and possibly others

### CODE REVIEW

**Review by pop4959 — DISMISSED** [review]

File: Essentials/src/com/earth2me/essentials/signs/SignEnchant.java [hunk 1]
```diff
@@ -28,21 +28,19 @@ protected boolean onSignCreate(final ISign sign, final User player, final String
...
             sign.setLine(2, "§c<enchant>");
             throw new SignException(tl("enchantmentNotFound"));
         }
-        int level;
-        try {
-            level = Integer.parseInt(enchantLevel[1]);
-        } catch (NumberFormatException ex) {
-            sign.setLine(2, "§c<enchant>");
-            throw new SignException(ex.getMessage(), ex);
+        if (enchantLevel.length == 2) {
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/signs/SignEnchant.java]: This logic opens up a weird validation corner case that probably doesn't matter but I'll still complain about....

`enchant:level:somejunk` is valid, but even if level > 1, level will always only be 1. This should be a validation error.

File: Essentials/src/com/earth2me/essentials/signs/SignEnchant.java [hunk 2]
```diff
@@ -70,18 +68,17 @@ protected boolean onSignInteract(ISign sign, User player, String username, IEsse
...
-        try {
-            level = Integer.parseInt(enchantLevel[1]);
-        } catch (NumberFormatException ex) {
-            level = enchantment.getMaxLevel();
+        int level = 1;
+        if (enchantLevel.length == 2) {
+            try {
+                level = Integer.parseInt(enchantLevel[1]);
+            } catch (NumberFormatException ex) {
+                level = enchantment.getMaxLevel();
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/signs/SignEnchant.java]: I don't like this default. I know you didn't do it, but this just feels clowny.

**Review by Ichbinjoe — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/signs/SignEnchant.java (hunk 2 above)
Comment (JRoy) [reply Essentials/src/com/earth2me/essentials/signs/SignEnchant.java]: Would you rather it be 1?

**Comment by JRoy** [comment]
Changes applied plus a question for @Ichbinjoe (check review comments)

File: Essentials/src/com/earth2me/essentials/signs/SignEnchant.java (hunk 2 above)
Comment (mdcfe) [reply Essentials/src/com/earth2me/essentials/signs/SignEnchant.java]: This currently retains the previous behaviour, where if the sign included a level but it was invalid, the sign would enchant at the max level. However, this case shouldn't ever get reached, as the sign is validated in `onSignCreate` above. If we've managed to get to this point, someone has deliberately bypassed validation, and I'd rather throw an error here than make assumptions about what the sign should do.

**Comment by JRoy** [comment]
@md678685 Improper enchantment levels now throw errors.

**Review by Ichbinjoe — APPROVED** [review]

**Review by mdcfe — APPROVED** [review]

**Pull request merged**